File: 2sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int sum =0;
        int[] output = new int[2];
        
        Map<Integer, Integer> map = new HashMap<>();
        for(int i=0;i<nums.length;i++) {
            if(map.containsKey(target-nums[i])) {
                output[0] = map.get(target-nums[i]);
                output[1] = i;
                return output;
            }
            map.put(nums[i] , i);
        }
        return output;
    }
}

File: 2Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> output;
        unordered_map<int , int> map;
        for(int i=0;i<nums.size();i++) {
            if(map.find(target-nums[i])!=map.end()) {
                output.push_back(map[target-nums[i]]);
                output.push_back(i);
                return output;
            }
            map[nums[i]] = i;
        }
        return output;

    }
};
